sql/core/src/test/resources/sql-tests/results/array.sql.out

```diff
@@ -124,8 +124,23 @@ struct<sort_array(boolean_array, true):array<boolean>,sort_array(tinyint_array,
 -- !query 8 output
 [true]	[1,2]	[1,2]	[1,2]	[1,2]	[9223372036854775808,9223372036854775809]	[1.0,2.0]	[1.0,2.0]	[2016-03-13,2016-03-14]	[2016-11-12 20:54:00.0,2016-11-15 20:54:00.0]
 
-
 -- !query 9
+select sort_array(array('b', 'd'), '1')
+-- !query 9 schema
+struct<>
+-- !query 9 output
+org.apache.spark.sql.AnalysisException
+cannot resolve 'sort_array(array('b', 'd'), '1')' due to data type mismatch: Sort order in second argument of sort_array() requires boolean type as non-null, however, it is ''1'' as string type.; line 1 pos 7
```

Comment:
FWIW, It seems it does not print it (`prettyName`) in the message as default in `ExpectsInputTypes`/`ImplicitCastInputTypes`.

```scala
s"argument ${idx + 1} requires ${expected.simpleString} type, " +
  s"however, '${child.sql}' is of ${child.dataType.simpleString} type."
```